File: 0547-number-of-provinces/0547-number-of-provinces.py
class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        res = 0
        visited = []

        def dfs(start: int) -> None:
            visited.append(start)
            for end in range(len(isConnected)):
                if isConnected[start][end] and end not in visited:
                    dfs(end)
        
        for i in range(len(isConnected)):
            if i not in visited:
                res += 1
                dfs(i)
        
        return res

File: 0547-number-of-provinces/0547-number-of-provinces.go
func dfs(visited map[int]bool, isConnected [][]int, startCity int) {
    for i := range isConnected[startCity] {
        if isConnected[startCity][i] == 1 && !visited[i] {
            visited[i] = true
            dfs(visited, isConnected, i)
        }
    }
}

func findCircleNum(isConnected [][]int) int {
    visited := map[int]bool{}
    ans := 0
    for i := range isConnected {
        if !visited[i] {
            dfs(visited, isConnected, i)
            ans += 1
        }
    }

    return ans
}